### PR TITLE
Hosting onboarding: Create developer specific site setup checklist

### PR DESCRIPTION
**Pull request by vykes-mac**
Related to https://github.com/Automattic/dotcom-forge/issues/2629

## Proposed Changes

* Add checklist task for developer accounts.

## Testing Instructions

- Apply patch D113245-code
- Select AT site
- Verify dev checklist is shown for a developer account.
- Complete each task and verify tasks are checked,

image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js [hunk 1]
```diff
@@ -344,6 +344,30 @@ export const getTask = (
 				isSkippable: true,
 			};
 			break;
+		case CHECKLIST_KNOWN_TASKS.INSTALL_CUSTOM_PLUGIN:
+			taskData = {
+				timing: 5,
+				title: translate( 'Install a custom plugin' ),
+				description: translate(
+					'Add new features to your site with plugins. Choose from thousands of free and premium plugins to make your site stand out.'
```
Comment (zaguiini) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/get-task.js]: This is good, but I wonder if we could add a note indicating that they can upload plugins too.

**Review by zaguiini — APPROVED** [review]
Looks good.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8046241

Thank you @vykes-mac for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8046241).

**Pull request merged**